zebra-network/src/address_book/tests/vectors.rs

```diff
@@ -110,3 +110,38 @@ fn address_book_peer_order() {
         Some(meta_addr2),
     );
 }
+
+/// Check that `reconnection_peers` skips addresses with IPs for which
+/// Zebra already has recently connected peers.
+#[test]
+fn reconnection_peers_skips_live_ip() {
+    let addr1 = "127.0.0.1:1".parse().unwrap();
+    let addr2 = "127.0.0.1:2".parse().unwrap();
+
+    let meta_addr1 = MetaAddr::new_responded(addr1, &PeerServices::NODE_NETWORK)
+        .into_new_meta_addr(
+            Instant::now(),
+            Utc::now().try_into().expect("will succeed until 2038"),
+        );
+    let meta_addr2 = MetaAddr::new_gossiped_meta_addr(
+        addr2,
+        PeerServices::NODE_NETWORK,
+        DateTime32::MIN.saturating_add(Duration32::from_seconds(1)),
+    );
+
+    // Regardless of the order of insertion, the most recent address should be chosen first
```

Comment:
```suggestion
    // The second address should be skipped because the first address has a
    // recent `last_response` time and the two addresses have the same IP.
```